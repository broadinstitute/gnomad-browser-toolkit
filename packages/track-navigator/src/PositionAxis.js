import PropTypes from 'prop-types'
import React from 'react'

const PositionAxis = ({ scalePosition, width }) => {
  const height = 15
  const numIntervals = Math.min(10, Math.floor(width / 90))

  const tickInterval = width / numIntervals
  const ticks = [...Array(numIntervals - 1)].map((_, i) => tickInterval * (i + 1))

  return (
    <svg height={height} width={width}>
      <line x1={0} y1={height} x2={width} y2={height} stroke="black" strokeWidth={2} />
      <g>
        <line x1={0} y1={height} x2={0} y2={height - 5} stroke="black" strokeWidth={2} />
        <text x={0} y={height - 7} textAnchor="start" style={{ fontSize: '10px' }}>
          {scalePosition.invert(0).toLocaleString()}
        </text>
      </g>
      {ticks.map(x => (
        <g key={x}>
          <line x1={x} y1={height} x2={x} y2={height - 5} stroke="black" strokeWidth={1} />
          <text x={x} y={height - 7} textAnchor="middle" style={{ fontSize: '10px' }}>
            {scalePosition.invert(x).toLocaleString()}
          </text>
        </g>
      ))}
      <g>
        <line x1={width} y1={height} x2={width} y2={height - 5} stroke="black" strokeWidth={2} />
        <text x={width} y={height - 7} textAnchor="end" style={{ fontSize: '10px' }}>
          {scalePosition.invert(width).toLocaleString()}
        </text>
      </g>
    </svg>
  )
}

PositionAxis.propTypes = {
  scalePosition: PropTypes.func.isRequired,
  width: PropTypes.number.isRequired,
}

export default PositionAxis
