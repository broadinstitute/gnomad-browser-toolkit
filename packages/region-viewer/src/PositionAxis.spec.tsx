import React from 'react'
import { expect, test, describe } from '@jest/globals'
import { render } from '@testing-library/react'

import { PositionAxis } from './PositionAxis'

describe('PositionAxis', () => {
  test('has no unexpected changes', () => {
    const scalePosition = (pos: number) => pos * 2
    scalePosition.invert = (pos: number) => pos / 2

    const { asFragment } = render(<PositionAxis scalePosition={scalePosition} width={800} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
