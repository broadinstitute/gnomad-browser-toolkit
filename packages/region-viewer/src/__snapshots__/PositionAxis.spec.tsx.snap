// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PositionAxis has no unexpected changes 1`] = `
<DocumentFragment>
  <svg
    height="15"
    width="800"
  >
    <line
      stroke="black"
      stroke-width="2"
      x1="0"
      x2="800"
      y1="15"
      y2="15"
    />
    <g>
      <line
        stroke="black"
        stroke-width="2"
        x1="0"
        x2="0"
        y1="15"
        y2="10"
      />
      <text
        style="font-size: 10px;"
        text-anchor="start"
        x="0"
        y="8"
      >
        0
      </text>
    </g>
    <g>
      <line
        stroke="black"
        stroke-width="1"
        x1="100"
        x2="100"
        y1="15"
        y2="10"
      />
      <text
        style="font-size: 10px;"
        text-anchor="middle"
        x="100"
        y="8"
      >
        50
      </text>
    </g>
    <g>
      <line
        stroke="black"
        stroke-width="1"
        x1="200"
        x2="200"
        y1="15"
        y2="10"
      />
      <text
        style="font-size: 10px;"
        text-anchor="middle"
        x="200"
        y="8"
      >
        100
      </text>
    </g>
    <g>
      <line
        stroke="black"
        stroke-width="1"
        x1="300"
        x2="300"
        y1="15"
        y2="10"
      />
      <text
        style="font-size: 10px;"
        text-anchor="middle"
        x="300"
        y="8"
      >
        150
      </text>
    </g>
    <g>
      <line
        stroke="black"
        stroke-width="1"
        x1="400"
        x2="400"
        y1="15"
        y2="10"
      />
      <text
        style="font-size: 10px;"
        text-anchor="middle"
        x="400"
        y="8"
      >
        200
      </text>
    </g>
    <g>
      <line
        stroke="black"
        stroke-width="1"
        x1="500"
        x2="500"
        y1="15"
        y2="10"
      />
      <text
        style="font-size: 10px;"
        text-anchor="middle"
        x="500"
        y="8"
      >
        250
      </text>
    </g>
    <g>
      <line
        stroke="black"
        stroke-width="1"
        x1="600"
        x2="600"
        y1="15"
        y2="10"
      />
      <text
        style="font-size: 10px;"
        text-anchor="middle"
        x="600"
        y="8"
      >
        300
      </text>
    </g>
    <g>
      <line
        stroke="black"
        stroke-width="1"
        x1="700"
        x2="700"
        y1="15"
        y2="10"
      />
      <text
        style="font-size: 10px;"
        text-anchor="middle"
        x="700"
        y="8"
      >
        350
      </text>
    </g>
    <g>
      <line
        stroke="black"
        stroke-width="2"
        x1="800"
        x2="800"
        y1="15"
        y2="10"
      />
      <text
        style="font-size: 10px;"
        text-anchor="end"
        x="800"
        y="8"
      >
        400
      </text>
    </g>
  </svg>
</DocumentFragment>
`;
